### ISSUE TITLE
System.Private.CoreLib.dll compilation with crossgen2 on armel leads to Error: Object reference not set to an instance of an object.

### ISSUE THREAD

**Issue by gbalykov** [dotnet/runtime#43035]
### Description

dotnet/runtime: 855e2961ba70d3ed8b094d230aaf655adac2b76e

armel:
```
`pwd`/corerun `pwd`/crossgen2/crossgen2.dll -r:`pwd`/* -r:`pwd`/crossgen2/* -O -o:`pwd`/System.Private.CoreLib.ni.dll `pwd`/System.Private.CoreLib.dll
mv System.Private.CoreLib.dll System.Private.CoreLib.dll.bak
mv System.Private.CoreLib.ni.dll System.Private.CoreLib.dll
`pwd`/corerun `pwd`/crossgen2/crossgen2.dll -r:`pwd`/* -r:`pwd`/crossgen2/* -O -o:`pwd`/System.Private.CoreLib.ni2.dll `pwd`/System.Private.CoreLib.dll
```

Output:
```
Error: Object reference not set to an instance of an object.
System.NullReferenceException: Object reference not set to an instance of an object.
   at System.Diagnostics.Tracing.EventSource.WriteEventWithRelatedActivityIdCore(Int32 eventId, Guid* relatedActivityId, Int32 eventDataCount, EventData* data)
   at System.Diagnostics.Tracing.EventSource.WriteEvent(Int32 eventId)
   at ILCompiler.PerfEventSource.JitMethodStop() in /home/runtime/src/coreclr/src/tools/aot/ILCompiler.ReadyToRun/Compiler/PerfEventSource.cs:line 119
   at ILCompiler.ReadyToRunCodegenCompilation.<ComputeDependencyNodeDependencies>b__23_0(DependencyNodeCore`1 dependency) in /home/runtime/src/coreclr/src/tools/aot/ILCompiler.ReadyToRun/Compiler/ReadyToRunCodegenCompilation.cs:line 507
   at ILCompiler.ReadyToRunCodegenCompilation.ComputeDependencyNodeDependencies(List`1 obj) in /home/runtime/src/coreclr/src/tools/aot/ILCompiler.ReadyToRun/Compiler/ReadyToRunCodegenCompilation.cs:line 513
   at ILCompiler.DependencyAnalysisFramework.DependencyAnalyzer`2.ComputeMarkedNodes()
   at ILCompiler.ReadyToRunCodegenCompilation.Compile(String outputFile)
   at ILCompiler.Program.Run()
   at ILCompiler.Program.InnerMain(CommandLineOptions buildOptions)
```

On x64 everything's fine

cc @alpencolt

**Comment by mangod9** [comment]
Appears to be tracing from here: https://github.com/dotnet/runtime/blob/0d62c3602c0214f95b7dc6ad165b5ddc9c1177db/src/coreclr/src/tools/aot/ILCompiler.ReadyToRun/Compiler/ReadyToRunCodegenCompilation.cs#L468. Perhaps there is a way to disable tracing and try on armel?

**Comment by gbalykov** [comment]
@mangod9 what is the correct way to disable tracing? I've had no success with undefining `FEATURE_PERFTRACING` or disabling `FeaturePerfTracing`, runtime doesn't build without them

**Comment by trylek** [comment]
@jkotas - this has been sitting around for a while, could you please advise what is the right fix here? I suspect you're one of the few people who have sufficiently broad context to make the call.

**Comment by jkotas** [comment]
This is likely armel specific codegen bug that just happened to be hit by crossgen2. @gbalykov Is it still a problem?

**Comment by gbalykov** [comment]
This was fixed by #43243, thanks!
